Czaple/Attractions/Screens/AttractionList.swift

```diff
@@ -48,30 +64,21 @@ class AttractionList: UITableViewController {
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: cellId, for: indexPath) as! AttractionCell
         
-        let attraction = attractions[indexPath.row + 1]
-//        let distance = currentLocation.distance(from: CLLocation(latitude: attraction.coordinate!.latitude, longitude: attraction.coordinate!.longitude)) / 1000
+        let attraction = attractionProvider.getAttractions()[indexPath.row + 1]
+        let distance = currentLocation.distance(from: CLLocation(latitude: attraction.coordinate!.latitude, longitude: attraction.coordinate!.longitude)) / 1000
         
         cell.photo.image = UIImage(named: attraction.name!)
         cell.title.text = attraction.name
-//        cell.distance.text = String(format:"%.2f", distance) + " km"
+        cell.distance.text = String(format:"%.2f", distance) + " km"
         cell.round.dropShadow()
         
         return cell
     }
     
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        let attractionVC = AttractionView()
-        attractionVC.attraction = attractions[indexPath.row + 1]
+        let attractionVC = AttractionView(boardAlert: boardAlert)
+        attractionVC.attraction = attractionProvider.getAttractions()[indexPath.row + 1]
```

Comment:
You could create a method in AttractionView that would change those attraction rather than exposing property :)